File: A2_Q1(b).py
'''
The average distance between two points on a 6 by 6 two-dimensional grid (no diagonal connections)
'''
rows,cols=(6,6)
arr=[]
dist=0
count=0
#creaing 6x6 2d array
for i in range (cols):
    col=[]
    for j in range (rows):
        col.append(0)
    arr.append(col)

for px in range(len(arr)):
    for py in range(len(arr[px])):
        for i in range (len(arr)):
            for j in range(len(arr[i])):
                dist=dist+abs(i-px)+abs(j-py)
                count+=1
print("\nTotal sum of the distance between two point in (6x6) grid is  : ",dist)
print("Total number of  the distances in (6x6) grid is : ",count)
print("\nAvg distance between two point in (6X6) grid is : ",float(dist/count))

File: A2_Q1(b).c
/*
#The average distance between two points on a 6 by 6 
two-dimensional grid (no diagonal connections)
*/
#include <stdio.h>
#include<stdlib.h>
int main()
{
	int dist=0;
	int count=0;
   	int i,j,px,py;
	for(px=0;px<6;px++)
	{
		for(py=0;py<6;py++)
		{
			for(i=0;i<6;i++)
			{
				for(j=0;j<6;j++)
				{
				dist=dist+abs(i-px)+abs(j-py);
				count++	;
				}	
			}	
		}
	}   
    
	printf("\nThe total sum of the distances between two points in (6x6)grid is  : %d ",dist);
    printf("\nThe total number of distances in (6x6)grid is :  %d\n",count);
    printf("\nThe average distances between two points in (6x6)grid is :  %f\n",(float)dist/(float)count);    
	
    return 0;
}
